src/components/Shop.jsx

```diff
@@ -16,27 +16,44 @@ import { Pagination } from "./Pagination";
 import { useDispatch, useSelector } from "react-redux";
 import { getCategories } from "../store/actions/globalAction";
 import { Link } from "react-router-dom/cjs/react-router-dom.min";
-import { getProducts } from "../store/actions/productAction";
+import { activePageSetter, getProducts } from "../store/actions/productAction";
 /* Responsive tasarım için bakman gereken yer product-container .*/
 export const Shop = () => {
   const { boxData, shopData } = data();
   const [loading, setLoading] = useState(true);
   const dispatch = useDispatch();
   const categoriesData = useSelector((store) => store.global.categories);
+  const productData = useSelector((store) => store.product.productList);
+  const productCount = useSelector((store) => store.product.totalProductCount);
+  console.log("productData", productData);
+  const sortByRating = categoriesData.sort((a, b) => b.rating - a.rating);
```

Comment:
sortByRating değişkeni doğrudan categoriesData üzerinde sort yaparak orijinal diziyi değiştiriyor. Dizinin orijinalliğini korumak için, sort işleminden önce bir kopyasını almalısın.
```
const sortByRating = [...categoriesData].sort((a, b) => b.rating - a.rating);
```.